src/esp/CMakeLists.txt

```
@@ -17,15 +17,595 @@ if(BUILD_WARNINGS_AS_ERRORS)
+  physics/RigidBase.h
+  physics/RigidObject.cpp
+  physics/RigidObject.h
+  physics/RigidStage.cpp
+  physics/RigidStage.h
+  physics/URDFImporter.cpp
+  physics/URDFImporter.h
+)
+
+if(BUILD_WITH_BULLET)
+
+  # Add bullet sources
+  list(
+    APPEND
+    physics_SOURCES
+    physics/bullet/BulletArticulatedLink.h
+    physics/bullet/BulletArticulatedObject.cpp
+    physics/bullet/BulletArticulatedObject.h
+    physics/bullet/BulletBase.cpp
+    physics/bullet/BulletBase.h
+    physics/bullet/BulletCollisionHelper.cpp
+    physics/bullet/BulletCollisionHelper.h
+    physics/bullet/BulletPhysicsManager.cpp
+    physics/bullet/BulletPhysicsManager.h
+    physics/bullet/BulletRigidObject.cpp
+    physics/bullet/BulletRigidObject.h
+    physics/bullet/BulletRigidStage.cpp
+    physics/bullet/BulletRigidStage.h
+    physics/bullet/BulletURDFImporter.cpp
+    physics/bullet/BulletURDFImporter.h
+    physics/bullet/objectWrappers/ManagedBulletArticulatedObject.h
+    physics/bullet/objectWrappers/ManagedBulletRigidObject.h
+  )
+
+  ## Enable physics profiling
+  #add_compile_definitions(BT_ENABLE_PROFILE=0)
+  #add_definitions(-DBT_ENABLE_PROFILE)
+
+endif() #BUILD_WITH_BULLET
+
+#################
+## nav
+set(nav_SOURCES nav/GreedyFollower.cpp nav/GreedyFollower.h nav/PathFinder.cpp
+                nav/PathFinder.h
+)
+
+#################
+## sensor
+set(
+  sensor_SOURCES
+  sensor/CameraSensor.cpp
+  sensor/CameraSensor.h
+  sensor/CubeMapSensorBase.cpp
+  sensor/CubeMapSensorBase.h
+  sensor/Sensor.cpp
+  sensor/Sensor.h
+  sensor/SensorFactory.cpp
+  sensor/SensorFactory.h
+  sensor/VisualSensor.cpp
+  sensor/VisualSensor.h
+  sensor/FisheyeSensor.cpp
+  sensor/FisheyeSensor.h
+  sensor/EquirectangularSensor.cpp
+  sensor/EquirectangularSensor.h
+  sensor/AudioSensor.cpp
+  sensor/AudioSensor.h
+  sensor/AudioSensorStubs.h
+)
+
+if(BUILD_WITH_CUDA)
+  list(APPEND sensor_SOURCES sensor/RedwoodNoiseModel.cpp sensor/RedwoodNoiseModel.h)
+  # CUDA kernel sources
+  set(noise_model_SOURCES sensor/RedwoodNoiseModel.cu sensor/RedwoodNoiseModel.cuh)
+endif()
+
+#################
+## sim
+set(
+  sim_SOURCES
+  sim/AbstractReplayRenderer.cpp
+  sim/AbstractReplayRenderer.h
+  sim/BatchPlayerImplementation.cpp
+  sim/BatchPlayerImplementation.h
+  sim/BatchReplayRenderer.cpp
+  sim/BatchReplayRenderer.h
+  sim/ClassicReplayRenderer.cpp
+  sim/ClassicReplayRenderer.h
+  sim/Simulator.cpp
+  sim/Simulator.h
+  sim/SimulatorConfiguration.cpp
+  sim/SimulatorConfiguration.h
+)
+
+#################
+## build library
+
+################################
+# Keeping gfx_batch a separate library
 add_subdirectory(gfx_batch)
-add_subdirectory(assets)
-add_subdirectory(metadata)
-add_subdirectory(io)
-add_subdirectory(scene)
-add_subdirectory(physics)
-add_subdirectory(nav)
-add_subdirectory(sensor)
-add_subdirectory(sim)
+##############################
+
+set(habsim_SOURCES ${core_SOURCES})
+list(
+  APPEND
+  habsim_SOURCES
+  ${geo_SOURCES}
```

Why not just this? Whitespace is treated as a list item separator; variables, unless put into quotes, get expanded as individual items also.

```suggestion
set(habsim_SOURCES 
  ${core_SOURCES}
  ${geo_SOURCES}
```

---

An artifact of my incremental development. I'll change it.